### PR TITLE
Fix new node failed to rejoin problem

### PR DESCRIPTION
**Pull request by ShengguangXiao**
## Description

Fix new node failed to rejoin problem.

1. Fix when boostrap failed to join testnet, and start to rejoin with retrieve history, but no history to retrieve and didn't add genesis block problem.
2. Fix duplicated rejoin procedure when in StartPoW function.

Run a local 20 nodes test, set COMM_SIZE to 8.

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

**Comment by ShengguangXiao** [comment]
Local test with some node always try to rejoin can run to Tx blocks 175.

File: src/libNode/PoWProcessing.cpp [hunk 1]
```diff
@@ -107,36 +107,7 @@ bool Node::StartPoW(const uint64_t& block_num, uint8_t ds_difficulty,
...
-              " rejoin");
-          RejoinAsNormal();
-        } else {
-          LOG_GENERAL(INFO, "Didn't get the latest DSBlock, what to do???");
-        }
-      }
-    };
-
-    DetachedFunction(1, func);
-  }
```
Comment (kaikawaliu) [inline src/libNode/PoWProcessing.cpp]: Why to remove this part? Is it causing problem?

Comment (ShengguangXiao) [reply src/libNode/PoWProcessing.cpp]: Yes, it will also cause rejoin, but now if wait for DS block timeout then trigger rejoin already.

Comment (KaustubhShamshery) [reply src/libNode/PoWProcessing.cpp]: This check here is for different purpose, if node is still mining for a long duration. The other thread only activates when node has finished mining.

Comment (ShengguangXiao) [reply src/libNode/PoWProcessing.cpp]: Then we should check the mining time, if the mining takes too long, then should stop mining. In my local test case, this piece of code make node always fetching latest Tx blocks, then it need another fix.

**Pull request merged**